### ISSUE TITLE
Possibilit of `dependentRequired` tag

### ISSUE THREAD

**Issue by victoraugustolls** [danielgtaylor/huma#296]
Is it possible to implement the `[dependentRequired](https://json-schema.org/draft/2020-12/json-schema-validation#name-dependentrequired)` validation tag? It is part of the json-schema 2020-12 that OpenAPI 3.1 supports!

**Comment by danielgtaylor** [comment]
@victoraugustolls yes I'd be open to adding support for it, and am happy to review a PR for it. It sounds like you'll want something like:

1. Update [`huma.Schema`](https://github.com/danielgtaylor/huma/blob/main/schema.go#L58) to add something like `DependentRequired map[string][]string`. Maybe add some pre-computed error messages for validation.
2. Update the [map validation](https://github.com/danielgtaylor/huma/blob/main/validate.go#L526-L674) to check if that new field is set in the schema, and if so look up which other fields are required as you iterate through each map entry.
3. Decide how it should be used, either by manually creating schemas or if there should be a high-level way to enable it using struct field tags. If struct tags, how would they work? Happy to discuss that here so we have a plan.
4. Add tests to keep coverage up.
5. Update any relevant [docs](https://github.com/danielgtaylor/huma/tree/main/docs/docs)

Thanks!

**Comment by victoraugustolls** [comment]
@danielgtaylor great! I will work on a pull request for it.

**Comment by victoraugustolls** [comment]
@danielgtaylor closing this issue!
